### PR TITLE
Purchase Details Mark II: Purchase Note cell

### LINKED ISSUE #858: Product Details UI: Purchase Details
On the new `ProductDetailsViewController`, implement the Purchase Details section (directly underneath the pricing and inventory section from #857). The section will be rendered differently depending on:
* if the product is Simple, Grouped, Affiliate, Virtual, or Variation
* If the data is available from `Yosemite` in `Product`/`ProductVariation`

(please see the `Product details` google sheet for more details)

Screen Shot 2019-04-09 at 2 28 06 PM
Screen Shot 2019-04-09 at 2 32 18 PM
Screen Shot 2019-04-09 at 2 34 39 PM
Screen Shot 2019-04-09 at 2 33 03 PM

Ref: #855 
Ref: #856
Ref: #857

### PR DESCRIPTION
**Pull request by mindgraffiti**
Closes #858.

This PR adds the Purchase note to Purchase Details if it exists. Previous PR #1009 contains screenshots for shipping detail and download cells.

A short purchase note
short-purchase-note

A long purchase note
long-purchase-note

### To test
- Check out and build this branch
- Go to your testing store's website
- Navigate to a product and add a product note (it's in the "Advanced" tab on the left)
- Update the product
- Create an order with that product
- Go back to the app and navigate to the product details view: Orders > Order Number > Fulfill order > Product list 
- Select product in the list
- Scroll to the bottom to see the purchase details information
- Select the "read more" button
- a fancy alert with the full text should appear, with a "dismiss" button at the bottom of the alert

Update release notes:

- [ ] If there are user facing changes, I have added an item to `RELEASE-NOTES.txt`.

### CODE REVIEW

File: WooCommerce/Classes/ViewModels/ProductDetailsViewModel.swift [hunk 1]
```diff
@@ -515,6 +546,24 @@ extension ProductDetailsViewModel {
...
+    ///
+    func configurePurchaseNote(_ cell: ReadMoreTableViewCell) {
+        cell.titleLabel?.text = NSLocalizedString("Purchase note",
+                                                  comment: "Product Details > Purchase Details > Purchase note cell title")
+
+        cell.bodyLabel?.text = cleanedPurchaseNote
+
+        let readMoreTitle = NSLocalizedString("Read more",
+                                              comment: "Read more of the purchase note. Only the first two lines of text are displayed.")
+        readMoreExpanded = false
```
Comment (ctarda) [inline WooCommerce/Classes/ViewModels/ProductDetailsViewModel.swift]: I am not sure what this flag does. It seems like it is initialised to false, and then set to false here as well, but I can not find anywhere else where the value is set or read. I was wondering if it's actually necessary?

File: WooCommerce/Classes/ViewModels/ProductDetailsViewModel.swift [hunk 2]
```diff
@@ -797,4 +846,8 @@ extension ProductDetailsViewModel {
         static let weightUnit = "woocommerce_weight_unit"
         static let dimensionUnit = "woocommerce_dimension_unit"
     }
+
+    enum Constants {
```
Comment (ctarda) [inline WooCommerce/Classes/ViewModels/ProductDetailsViewModel.swift]: If the removal of the last new line is done using the method extracted to the helper class, maybe this is not necessary

File: WooCommerce/Classes/ViewModels/ProductDetailsViewModel.swift [hunk 3]
```diff
@@ -27,12 +31,33 @@ final class ProductDetailsViewModel {
...
+    /// - stripped of HTML
+    /// - no ending newline character
+    /// - cannot be a lazy var because it's a computed property
+    ///
+    var cleanedPurchaseNote: String? {
+        guard var cleanedString = product.purchaseNote?.strippedHTML else {
+            return nil
+        }
+
+        let lastChar = cleanedString.suffix(1)
```
[the remark this answers is not in the document]
Comment (mindgraffiti) [reply WooCommerce/Classes/ViewModels/ProductDetailsViewModel.swift]: Good catch! I missed a step.

File: WooCommerce/Classes/ViewModels/ProductDetailsViewModel.swift (hunk 1 above)
Comment (mindgraffiti) [reply WooCommerce/Classes/ViewModels/ProductDetailsViewModel.swift]: Good eye! That's an artifact from trying out WordPressUI's "expand selected row" method in a previous commit. Looks like I missed some pieces after trying it out.

**Comment by mindgraffiti** [comment]
Forgot to mention this the PR request "To Test" section. I benchmarked the old Strip HTML method that WordPress uses (a regex method) and compared it to the new `NSAttributedString` method that converts HTML data into an attributed string, then returns the plain string from the attributed one.

You can test them yourself by checking out this commit https://github.com/woocommerce/woocommerce-ios/pull/1016/commits/b6d91312aa1c0b3c30853e0661ee198bd7bb49c0 in this PR (titled "Benchmark tests for methods that strip HTML" )

These were my results, for 3 separate tests, in seconds:

```
Time elapsed for stripped HTML in purchase note: 0.023846030235290527 s.
Time elapsed for old fashioned strip HTML: 9.09566879272461e-05 s.

Time elapsed for stripped HTML in purchase note: 0.022452950477600098 s.
Time elapsed for old fashioned strip HTML: 1.800060272216797e-05 s.

Time elapsed for old fashioned strip HTML: 2.110004425048828e-05 s.
Time elapsed for stripped HTML in purchase note: 0.022012948989868164 s.
```

**Pull request merged**